facia-press/app/frontpress/FapiFrontPress.scala

```diff
@@ -148,29 +150,61 @@ trait FapiFrontPress extends Logging {
     pressFuture
   }
 
-  def generateCollectionJsonFromFapiClient(collectionId: String)(implicit executionContext: ExecutionContext): Response[PressedCollection] =
+  private def putPressedPage(path: String, pressedFront: PressedPage, pressedType: PressedPageType): Unit = {
+    val json: String = Json.stringify(Json.toJson(pressedFront))
+
+    val metric: SamplerMetric = pressedType match {
+      case FullType => FaciaPressMetrics.FrontPressContentSize
+      case LiteType => FaciaPressMetrics.FrontPressContentSizeLite
+    }
+
+    metric.recordSample(json.getBytes.length, new DateTime())
+    putPressedJson(path, json, pressedType)
+  }
+
+  def generateCollectionJsonFromFapiClient(collectionId: String)(implicit executionContext: ExecutionContext): Response[PressedCollectionVersions] = {
     for {
       collection <- FAPI.getCollection(collectionId)
       curated <- getCurated(collection)
       backfill <- getBackfill(collection)
       treats <- getTreats(collection)
     } yield {
       val doNotTrimContainerOfTypes = Seq("nav/list")
-      val maxStories: Int = doNotTrimContainerOfTypes
+      val storyCountTotal = curated.length + backfill.length
+      val storyCountMax: Int = doNotTrimContainerOfTypes
         .contains(collection.collectionConfig.collectionType)
-        .toOption(curated.length + backfill.length)
-        .getOrElse(Configuration.facia.collectionCap)
-      val trimmedCurated = curated.take(maxStories)
-      val trimmedBackfill = backfill.take(maxStories - trimmedCurated.length)
-      PressedCollection.fromCollectionWithCuratedAndBackfill(
-        collection,
-        trimmedCurated,
-        trimmedBackfill,
-        treats
+        .toOption(storyCountTotal)
+        .getOrElse(Math.min(Configuration.facia.collectionCap, storyCountTotal))
+      val storyCountVisible = Container.storiesCount(CollectionConfig.make(collection.collectionConfig), curated ++ backfill).getOrElse(storyCountMax)
+      val hasMore = storyCountVisible < storyCountMax
```

Comment:
storyCountVisible should probably be min(storyCountVisible, storyCountMax)

Reply:
`storyCountVisible` is either the number of stories necessary to fill the expected layout or `storyCountMax`.
In theory no container layout requires more than the max cap